docs/virtual_machines/istio_service_mesh.md

```diff
@@ -0,0 +1,82 @@
+# Istio service mesh
+
+Service mesh allows to monitor, visualize and control traffic between microservices.
+Kubevirt supports running VMs as a part of Istio service mesh.
+
+## Limitations
+
+- Istio service mesh is only supported with a pod network masquerade binding.
+
+- Istio uses a [list of ports](https://istio.io/latest/docs/ops/deployment/requirements/#ports-used-by-istio) for it's own functionality, these ports must not
+  be explicitly specified in a VMI interface.
+
+## Prerequisites
+
+This guide assumes that Istio is already deployed and uses Istio CNI Plugin. See [Istio documentation](https://istio.io/latest/docs/) for more information.
+
+## Create a VirtualMachineInstance with enabled Istio proxy injecton
+
+If your cluster uses Multus to manage CNI, the following NetworkAttachmentDefinition is required in the application namespace:
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+```
+
+The example below specifies a VMI with masquerade network interface and annotation instructing
+Istio webhook to inject the proxy container.
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  annotations:
+    "sidecar.istio.io/inject": "true"
+  labels:
+    app: vmi-istio
+    version: v1
+  name: vmi-istio
+spec:
+  domain:
+    devices:
+      interfaces:
+        - name: default
+          masquerade: {}
+          ports:
+            - port: 8080
+            - port: 80
+      disks:
+        - disk:
+            bus: virtio
+          name: containerdisk
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+    - name: default
+      pod: {}
+  terminationGracePeriodSeconds: 0
+  volumes:
+    - name: containerdisk
+      containerDisk:
+        image: registry:5000/kubevirt/fedora-cloud-container-disk-demo:devel
+```
+
+**Note:** Make sure that the target namespace has label enabling automatic sidecar injection: `istio-injection=enabled`
+
+## Verification
```

Comment:
@phoracek Not sure if this section is really helpful, wdyt?

Reply:
I am all for verification steps. It would help bisect any bug that may happen here. +1

Reply:
Added